src/app/cardsView/epic/epic.component.ts

```diff
@@ -5,28 +5,29 @@ import { Component } from '@angular/core';
 import {ActivatedRoute} from "@angular/router";
 import {ParentCard} from "../../card/card";
 import {EpicService} from "../services/epic/epic.service";
+import {BasePaginateCardComponent} from "../base-paginate-card.component";
 
 
 @Component({
   selector: 'epic',
   templateUrl: '../base-view.component.html',
   styleUrls: [ '../base-view.component.css' ]
 })
-export class EpicComponent {
-  private cards: ParentCard[];
+export class EpicComponent extends BasePaginateCardComponent {
   private key: number;
 
   constructor(private route: ActivatedRoute, private service: EpicService) {
+    super();
   }
 
-  ngOnInit() {
+  getPage(page: number) {
+    let keys: string[] =[];
+
     this.route.params
       .map(params => [params['projectKey'], params['epicKey']])
-      .subscribe(keys => {
-        this.key = keys[1];
-        this.service
-          .get(keys[0], keys[1])
-          .subscribe(cards => this.cards = cards);
-      });
+      .subscribe(ks => keys = ks);
+
+    // TODO Wait for this.key assignment in this.service.getPage
+    this._getPage(page, this.service.getPage(page, this.perPage, keys[0], keys[1]));
```

Comment:
Ktoś się orientuje jak ogarnąć najlepiej te observable tutaj?
Metoda `_getPage()` potrzebuje `Observable<coś tam>` z `service.getPage()`, ale klucze potrzebne w metodzie są ustawiane również z observable z paramsów route'a...
Kminiłem nad `combineLatest() / map()`, ale nie ogarnąłem jak to zrobić. Ostatecznie można by delaya jakeigoś obadać, ale też nie wykminiłem. Ktoś coś? Możliwe, że można do tego jakoś zupełnie inaczej podejść, może coś innego można zmienić...